src/main/scala/com/typesafe/sbt/SbtProguard.scala

```diff
@@ -26,19 +32,30 @@ object SbtProguard extends AutoPlugin {
     proguardVersion := "7.0.0",
     proguardDirectory := crossTarget.value / "proguard",
     proguardConfiguration := proguardDirectory.value / "configuration.pro",
-    artifactPath := proguardDirectory.value / (artifactPath in packageBin in Compile).value.getName,
+    artifactPath := proguardDirectory.value / ( Compile / packageBin / artifactPath).value.getName,
     managedClasspath := Classpaths.managedJars(configuration.value, classpathTypes.value, update.value),
-    proguardBinaryDeps := getAllBinaryDeps.value,
-    proguardInputs := (fullClasspath in Runtime).value.files,
-    proguardLibraries := proguardBinaryDeps.value filterNot proguardInputs.value.toSet,
+    proguardBinaryDeps := {
+      (Compile / libraryDependencies).value match {
+        case analysis: Analysis =>
+          analysis.relations.allLibraryDeps.collect { case vf: PathBasedFile =>
+            vf.toPath.toFile
+          }.toSeq
+      }
+    },
+    proguardInputs := {
+      assembly.value
+      Seq((assembly / assemblyOutputPath).value)
+    },
```

Comment:
Is there a reason for depending on sbt-assembly and always going through an assembly first? Just using the classpath seems like a simpler default, and pre-assembly can be added when needed?

Reply:
I couldn't get it to work by using the classpath. My understanding (correct me if i'm wrong) is that Proguard can only work on fat jars. But maybe I was doing something wrong

Reply:
Just tried it again and it's still running into errors. It's possible that the task keys this was using before have changed between versions

Reply:
Ok. Since there are other changes, both inputs and libraries, could be good to confirm what's going on here. It should certainly work as before, with a classpath of inputs rather than a fat jar.

Reply:
I'm really not a proguard expert but would also think that sbt-assembly shouldn't be mandated. 
It wouldn't be the end of the world and if there's no other way I'd rather merge this PR. 

However: I just tried your branch and reverted to the old `proguardInputs := (fullClasspath in Runtime).value.files` and removed all sbt-assembly related settings/dependency declarations. The included test projects work just fine (two of which do not use sbt-assembly themselves), and so does our internal build (which uses sbt-native-packager). 

Can you clarify what kind of errors you're running into? Maybe those are actually unrelated to assembly?

Reply:
That's odd. None of the tests work for me when I try that. This is the error I get:

```
[info] [info] Proguard command:
[info] [info] java -Xmx256M -cp /Users/sbly/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/guardsquare/proguard-base/7.0.0/proguard-base-7.0.0.jar:/Users/sbly/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/guardsquare/proguard-core/7.0.0/proguard-core-7.0.0.jar:/Users/sbly/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/google/code/gson/gson/2.8.5/gson-2.8.5.jar:/Users/sbly/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jetbrains/kotlin/kotlin-stdlib/1.3.31/kotlin-stdlib-1.3.31.jar:/Users/sbly/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jetbrains/kotlin/kotlin-stdlib-common/1.3.31/kotlin-stdlib-common-1.3.31.jar:/Users/sbly/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jetbrains/kotlinx/kotlinx-metadata-jvm/0.1.0/kotlinx-metadata-jvm-0.1.0.jar:/Users/sbly/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jetbrains/annotations/13.0/annotations-13.0.jar proguard.ProGuard -injars "/private/var/folders/34/jl47bvtd2_14y99v6gx4k0bm0000gt/T/sbt_d73d4659/target/scala-2.12/classes" -injars "/Users/sbly/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.12.3/scala-library-2.12.3.jar" -libraryjars "/Users/sbly/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.12.3/scala-library-2.12.3.jar" -libraryjars "/Users/sbly/Library/Java/JavaVirtualMachines/corretto-11.0.12/Contents/Home" -outjars "/private/var/folders/34/jl47bvtd2_14y99v6gx4k0bm0000gt/T/sbt_d73d4659/target/scala-2.12/proguard/simple_2.12-0.1.0-SNAPSHOT.jar" -dontoptimize -dontnote -dontwarn -ignorewarnings -keep public class Test {
[info] [info]     public static void main(java.lang.String[]);
[info] [info] }
[info] [info] ProGuard, version 7.0.0
[info] [error] Error: The same input jar [/Users/sbly/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.12.3/scala-library-2.12.3.jar] is specified twice.
[info] [error] java.lang.RuntimeException: Proguard failed with exit code [1]
```
Depending on what other settings I tweak, I encounter other errors, but I can never get it working without assembly. If you can, that's great. Maybe my local setup is somehow broken

Reply:
update: we removed sbt-assembly for the final version of this PR